### ISSUE TITLE
Stochastic gradient descent performs variational inference, converges to limit cycles for deep networks

### ISSUE THREAD

**Issue by richardtomsett** [dais-ita/interpretability-papers#16]
[Stochastic gradient descent performs variational inference, converges to limit cycles for deep networks](https://arxiv.org/abs/1710.11029)
Stochastic gradient descent (SGD) is widely believed to perform implicit regularization when used to train deep neural networks, but the precise manner in which this occurs has thus far been elusive. We prove that SGD minimizes an average potential over the posterior distribution of weights along with an entropic regularization term. This potential is however not the original loss function in general. So SGD does perform variational inference, but for a different loss than the one used to compute the gradients. Even more surprisingly, SGD does not even converge in the classical sense: we show that the most likely trajectories of SGD for deep networks do not behave like Brownian motion around critical points. Instead, they resemble closed loops with deterministic components. We prove that such "out-of-equilibrium" behavior is a consequence of the fact that the gradient noise in SGD is highly non-isotropic; the covariance matrix of mini-batch gradients has a rank as small as 1% of its dimension. We provide extensive empirical validation of these claims, proven in the appendix.

Bibtex:

@misc{1710.11029,
Author = {Pratik Chaudhari and Stefano Soatto},
Title = {Stochastic gradient descent performs variational inference, converges to limit cycles for deep networks},
Year = {2017},
Eprint = {arXiv:1710.11029},
}
